src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,168 @@
+                            System.out.printf("\t\t%s\n", task);
+                            System.out.printf("\tNow you have %d tasks in the list.\n", todo.size());
+
+                            break;
+                        } catch (DukeException err) {
+                            System.out.println(err.getMessage());
+                            break;
+                        } catch (DateTimeParseException err) {
+                            System.out.println("\tPlease write your data in d/m/yyyy T format");
+                            break;
+                        }
+                    case ("event"):
+
+                        try {
+                            task = Parser.parse_event(input);
+                            todo.addTask(task);
+                            System.out.println("\tGot it. I've added this task:");
+                            System.out.printf("\t\t%s\n", task);
+                            System.out.printf("\tNow you have %d tasks in the list.\n", todo.size());
+
+                            break;
+                        } catch (DukeException err) {
+                            System.out.println(err.getMessage());
+                            break;
+                        }
+
+                    case("delete"):
+                        try {
+                            idx = Parser.parse_delete(input);
+                            System.out.println("\tNoted. I've removed this task:");
+                            System.out.printf("\t\t%s\n", todo.deleteTask(idx));
+                            System.out.printf("\tNow you have %d tasks in the list.\n", todo.size());
+                            break;
+                        } catch (DukeException err) {
+                            System.out.println(err.getMessage());
+                            break;
+                        }
+                    case("find"):
+                        try {
+                            keyword = Parser.parse_find(input);
+                            System.out.println("\tHere are the matching tasks in your list:");
+                            todo.find(keyword);
+                            break;
+                        } catch (DukeException err) {
+                            System.out.println(err.getMessage());
+                            break;
+                        }
+
+                    default:
+                        throw new DukeException("\tSorry, I did not understand the command!");
+
+                }
+                System.out.printf("\t%s\n", divider);
+                storage.writeFile(todo.getList());
+            } catch (DukeException err) {
+                System.out.println(err.getMessage());
+                System.out.printf("\t%s\n", divider);
+            } catch (IOException err) {
+                System.out.println(err.getMessage());
+                System.out.printf("\t%s\n", divider);
+            }
+        }
+    }
```

Comment:
1. It might be better to have a separate class to handle commands.
2. Abstract out the printing of tasks-related messages to the methods inside Task.java themselves.